packages/embark-code-runner/src/index.ts

```diff
@@ -52,101 +50,57 @@ export default class CodeRunner {
     });
   }
 
-  private fireEmbarkJSEvents(code: string) {
-    const regexRegister = /EmbarkJS\.(.*)\.registerProvider/gm;
-    const regexSet = /EmbarkJS\.(.*)\.setProvider/gm;
-    let matches = regexRegister.exec(code);
-    if (matches) {
-      let [, provider] = matches;
-      provider = provider.toLowerCase();
-      this.providerStates[`${provider}:${ProviderEventType.ProviderRegistered}`] = true;
-      this.events.emit(`runcode:${provider}:${ProviderEventType.ProviderRegistered}`);
-    }
-    matches = regexSet.exec(code);
-    if (matches) {
-      let [, provider] = matches;
-      provider = provider.toLowerCase();
-      this.providerStates[`${provider}:${ProviderEventType.ProviderSet}`] = true;
-      this.events.emit(`runcode:${provider}:${ProviderEventType.ProviderSet}`);
-    }
-  }
-
   private registerEvents() {
-  this.events.on("runcode:register", this.registerVar.bind(this));
-
-  this.events.on("runcode:init-console-code:updated", (code: string, cb: Callback<null>) => {
-    this.evalCode(code, (err, _result) => {
-      if (err) {
-        this.logger.error("Error running init console code: ", err.message || err);
-      }
-      this.fireEmbarkJSEvents(code);
-      cb();
-    });
-  });
+    this.events.on("runcode:register", this.registerVar.bind(this));
+  }
 
-  this.events.on("runcode:embarkjs-code:updated", (code: string, cb: Callback<any>) => {
-    this.evalCode(code, (err, _result) => {
-      if (err) {
-        this.logger.error("Error running embarkjs code: ", err.message || err);
-      }
-      this.fireEmbarkJSEvents(code);
-      cb();
+  private registerCommands() {
+    this.events.setCommandHandler("runcode:getContext", (cb) => {
+      cb(this.vm.options.sandbox);
     });
-  });
-}
+    this.events.setCommandHandler("runcode:eval", this.evalCode.bind(this));
+    this.events.setCommandHandler("runcode:embarkjs:reset", this.resetEmbarkJS.bind(this));
+    this.events.on("contractsDeployed", this.resetEmbarkJS.bind(this));
+
+    // register listeners for when EmbarkJS runs registerProvider through the console.
+    // For example, when `EmbarkJS.Storage.registerProvider(...)` is run through the console,
+    // emit the `runcode:storage:providerRegistered` event, and fire any requests attached to it
+    Object.keys(EmbarkJS)
+      .filter((propName) => EmbarkJS[propName].hasOwnProperty("registerProvider"))
+      .forEach((providerName) => {
+        this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderRegistered);
+      });
```

Comment:
Just a little note: When using fat arrow functions that make use of only one parameter, we can omit the additional parenthesis:

```
Object.keys(EmbarkJS)
  .filter(propName => ...)
  .forEach(providerName => ...)
```

Reply:
I believe the tslint won't allow that, with our current settings.

Reply:
Ah, I keep forgetting..

Reply:
It's ok to generate the listener but I believe the code to emit the events has been removed.